src/oge/helpers.py

```diff
@@ -204,6 +204,252 @@ def assign_ba_code_to_plant(df: pd.DataFrame, year: int) -> pd.DataFrame:
     return df
 
 
+def assign_fleet_to_subplant_data(
+    subplant_data: pd.DataFrame,
+    plant_attributes_table: pd.DataFrame,
+    primary_fuel_table: pd.DataFrame,
+    ba_col: str = "ba_code",
+    primary_fuel_col: str = "subplant_primary_fuel",
+    fuel_category_col: str = "fuel_category",
+    other_attribute_cols: list[str] = [],
+    drop_primary_fuel_col: bool = True,
+) -> pd.DataFrame:
+    """Assigns a BA code and fuel category to each subplant in order to facilitate
+    aggregating the data to the fleet level.
+
+    When assigning a primary fuel/fuel category, the general options we should follow
+    are:
+        - For applying to CEMS data for the residual hourly profile calculation:
+            - primary_fuel_col = "subplant_primary_fuel_from_capacity_mw"
+            - fuel_category_col = "fuel_category_eia930"
+            - Notes: this matches how generators would likely be classified for 930
+            reporting
+        - For applying to monthly EIA-923 data to shape:
+            - primary_fuel_col = "subplant_primary_fuel_from_capacity_mw"
+            - fuel_category_col = "fuel_category"
+            - Notes: This means that we are assigning our flat profiles for all the
```

Comment:
flat -> fleet

Reply:
This is actually meant to be "flat" meaning the flat profiles we create as one of the shape options. But I'll add quotes to flat so we know it is a referenced name

Reply:
I saw that in the `impute_hourly_profiles` file that I reviewed later. I left a comment [there](https://github.com/singularity-energy/open-grid-emissions/pull/395#discussion_r1857481182).